src/System.Runtime.Extensions/src/System/IO/Path.Windows.cs

```diff
@@ -41,145 +41,58 @@ public static string GetFullPath(string path)
             // This is because the nulls will signal the end of the string to Win32 and therefore have
             // unpredictable results. Other invalid characters we give a chance to be normalized out.
             if (path.IndexOf('\0') != -1)
                 throw new ArgumentException(SR.Argument_InvalidPathChars, nameof(path));
 
-            // Toss out paths with colons that aren't a valid drive specifier.
-            // Cannot start with a colon and can only be of the form "C:" or "\\?\C:".
-            // (Note that we used to explicitly check "http:" and "file:"- these are caught by this check now.)
-            int startIndex = PathInternal.PathStartSkip(path);
-            bool isExtended = path.Length >= PathInternal.ExtendedPathPrefix.Length + startIndex
-                && path.IndexOf(PathInternal.ExtendedPathPrefix, startIndex, PathInternal.ExtendedPathPrefix.Length, StringComparison.Ordinal) >= 0;
-
-            if (isExtended)
+            if (PathInternal.IsExtended(path))
             {
-                startIndex += PathInternal.ExtendedPathPrefix.Length;
+                // We can't really know what is valid for all cases of extended paths.
+                //
+                //  - object names can include other characters as well (':', '/', etc.)
+                //  - even file objects have different rules (pipe names can contain most characters)
+                //
+                // As such we will do no further analysis of extended paths to avoid blocking known and unknown
+                // scenarios as well as minimizing compat breaks should we block now and need to unblock later.
+                return path;
             }
 
-            // Move past the colon
-            startIndex += 2;
-
-            if ((path.Length > 0 && path[0] == VolumeSeparatorChar)
-                || (path.Length >= startIndex && path[startIndex - 1] == VolumeSeparatorChar && !PathInternal.IsValidDriveChar(path[startIndex - 2]))
-                || (path.Length > startIndex && path.IndexOf(VolumeSeparatorChar, startIndex) != -1))
+            bool isDevice = PathInternal.IsDevice(path);
+            if (!isDevice)
             {
-                throw new NotSupportedException(SR.Argument_PathFormatNotSupported);
-            }
+                // Toss out paths with colons that aren't a valid drive specifier.
+                // Cannot start with a colon and can only be of the form "C:".
+                // (Note that we used to explicitly check "http:" and "file:"- these are caught by this check now.)
+                int startIndex = PathInternal.PathStartSkip(path);
 
-            if (isExtended)
-            {
-                // If the path is in extended syntax, we don't need to normalize, but we still do some basic validity checks
-                if (!ValidateExtendedPath(path))
+                // Move past the colon
+                startIndex += 2;
```

Comment:
Note that the ADS syntax I'm still explicitly rejecting. If you want it you need to use a device path of some sort. We can revisit opening up non device paths, but so few know about ADS it seemed prudent to keep it an "advanced" feature.